modules/apps/portal-cache/portal-cache-ehcache-impl/src/main/java/com/liferay/portal/cache/ehcache/internal/BaseEhcachePortalCacheManager.java

```diff
@@ -400,6 +408,16 @@ public void removedService(
 	@Reference
 	protected Props props;
 
+	private long _getShardedCacheCompanyId(String portalCacheName) {
+		int indexOf = portalCacheName.lastIndexOf(StringPool.UNDERLINE);
+
+		if (indexOf == -1) {
+			return 0L;
+		}
+
+		return GetterUtil.getLong(portalCacheName.substring(indexOf + 1));
+	}
+
```

Comment:
@marianoalvarosaiz, I think `_getShardedCacheCompanyId` may be not suitable for this. Refer to `ShardedEhcachePortalCache#getEhcache()`, the key is the real companyId. If portalCacheName included UNDERLINE, for example, "A_BC", for the case, it will use ShardedEhcachePortalCache and this is not right.